servers/rendering/rasterizer_rd/shaders/scene_high_end.glsl

```diff
@@ -1511,371 +1023,896 @@ void sdfgi_process(uint cascade, vec3 cascade_pos, vec3 cam_pos, vec3 cam_normal
+					vec3 pos = splane.xyz + tangent * disk.x + bitangent * disk.y;
+
+					pos = normalize(pos);
+					vec4 uv_rect = lights.data[light_idx].atlas_rect;
+
+					if (pos.z >= 0.0) {
+						pos.z += 1.0;
+						uv_rect.y += uv_rect.w;
+					} else {
+						pos.z = 1.0 - pos.z;
+					}
+
+					pos.xy /= pos.z;
 
-		// Compute weight
+					pos.xy = pos.xy * 0.5 + 0.5;
+					pos.xy = uv_rect.xy + pos.xy * uv_rect.zw;
+					shadow += textureProj(sampler2DShadow(shadow_atlas, shadow_sampler), vec4(pos.xy, z_norm, 1.0));
+				}
 
-		vec3 probe_pos = vec3(probe_posi);
-		vec3 probe_to_pos = cascade_pos - probe_pos;
-		vec3 probe_dir = normalize(-probe_to_pos);
+				shadow /= float(scene_data.penumbra_shadow_samples);
 
-		vec3 trilinear = vec3(1.0) - abs(probe_to_pos);
-		float weight = trilinear.x * trilinear.y * trilinear.z * max(0.005, dot(cam_normal, probe_dir));
+			} else {
+				//no blockers found, so no shadow
+				shadow = 1.0;
+			}
+		} else {
+			splane.xyz = normalize(splane.xyz);
+			vec4 clamp_rect = lights.data[light_idx].atlas_rect;
 
-		// Compute lightprobe occlusion
+			if (splane.z >= 0.0) {
+				splane.z += 1.0;
 
-		if (sdfgi.use_occlusion) {
-			ivec3 occ_indexv = abs((sdfgi.cascades[cascade].probe_world_offset + probe_posi) & ivec3(1, 1, 1)) * ivec3(1, 2, 4);
-			vec4 occ_mask = mix(vec4(0.0), vec4(1.0), equal(ivec4(occ_indexv.x | occ_indexv.y), ivec4(0, 1, 2, 3)));
+				clamp_rect.y += clamp_rect.w;
 
-			vec3 occ_pos = clamp(cascade_pos, probe_pos - sdfgi.occlusion_clamp, probe_pos + sdfgi.occlusion_clamp) * sdfgi.probe_to_uvw;
-			occ_pos.z += float(cascade);
-			if (occ_indexv.z != 0) { //z bit is on, means index is >=4, so make it switch to the other half of textures
-				occ_pos.x += 1.0;
+			} else {
+				splane.z = 1.0 - splane.z;
 			}
 
-			occ_pos *= sdfgi.occlusion_renormalize;
-			float occlusion = dot(textureLod(sampler3D(sdfgi_occlusion_cascades, material_samplers[SAMPLER_LINEAR_CLAMP]), occ_pos, 0.0), occ_mask);
+			splane.xy /= splane.z;
 
-			weight *= max(occlusion, 0.01);
+			splane.xy = splane.xy * 0.5 + 0.5;
+			splane.z = (shadow_len - lights.data[light_idx].shadow_bias) * lights.data[light_idx].inv_radius;
+			splane.xy = clamp_rect.xy + splane.xy * clamp_rect.zw;
+			splane.w = 1.0; //needed? i think it should be 1 already
+			shadow = sample_pcf_shadow(shadow_atlas, lights.data[light_idx].soft_shadow_scale * scene_data.shadow_atlas_pixel_size, splane);
 		}
 
-		// Compute lightprobe texture position
+#ifdef LIGHT_TRANSMITTANCE_USED
+		{
+			vec4 clamp_rect = lights.data[light_idx].atlas_rect;
 
-		vec3 diffuse;
-		vec3 pos_uvw = diffuse_posf;
-		pos_uvw.xy += vec2(offset.xy) * sdfgi.lightprobe_uv_offset.xy;
-		pos_uvw.x += float(offset.z) * sdfgi.lightprobe_uv_offset.z;
-		diffuse = textureLod(sampler2DArray(sdfgi_lightprobe_texture, material_samplers[SAMPLER_LINEAR_CLAMP]), pos_uvw, 0.0).rgb;
+			//redo shadowmapping, but shrink the model a bit to avoid arctifacts
```

Comment:
Did you redo shadowmapping?

Reply:
This comment is old. I did not intend to change anything about how shadowmapping worked, as this is a refactor.

The diff is unfortunately completely incomprehensible. You need to view them side by side or use a different diff viewer combined with an "ignore whitespace" option that has a concept of "moved to"

Reply:
Remove the comment if unused?